### PR TITLE
Find config file closest to the editor file

### PR DESCRIPTION
**Pull request by emoryy**
Related to issue [#12 ](https://github.com/binhums/linter-ember-template/issues/12)

### CODE REVIEW

File: lib/linter-ember-template.js [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+function getConfigFileFromDirectory(directory) {
+  return directory.getEntriesSync().find(function(entry) {
+    return entry.isFile() && entry.getBaseName() === configFileName;
+  });
+}
+
+function findConfigFileForFilePath(editorFilePath) {
+  let currentDir = new File(editorFilePath).getParent();
+  let foundConfig;
+  while (!(foundConfig = getConfigFileFromDirectory(currentDir)) && !currentDir.isRoot()) {
```
Comment (binhums) [inline lib/linter-ember-template.js]: This is a bit difficult to parse. I discourage variable assignment in the predicate for `if` or `while` statements. Conditional logic here can get messy too so I'd prefer being explicit about our intentions. For example something like

    let currentDir = new File(editorFilePath).getParent();

    while(true) {

      let configFile = getConfigFileFromDirectory(currentDir);
      if (configFile) {
        return configFile.getPath();
      }

      if (currentDir.isRoot()) {
        return null; // or maybe just break;
      }
      
      currentDir = currentDir.getParent();
    }

File: lib/linter-ember-template.js [hunk 2]
```diff
@@ -18,7 +36,7 @@ module.exports = {
       lint: function(textEditor) {
         return new Promise(function(resolve, reject) {
           let options = {
-            configPath: atom.project.getPaths()[0] + "/.template-lintrc"
+            configPath: findConfigFileForFilePath(textEditor.getPath())
```
Comment (binhums) [inline lib/linter-ember-template.js]: A possible problem here is that we're being explicit about the fact that we might not find a config file.

The existing code will provide a path to a file that doesn't exist if that is the case, and that's not ideal. But providing a `configPath` param with null is probably also not ideal. It's not a major problem because `ember-template-lint` handles that case.

I suspect we should only specify the `configPath` option if we have one we want to use. This way we let `ember-template-lint` implement its own fallback. As the code stands [this will choose the template lint config of the directory of the currently running process](https://github.com/rwjblue/ember-template-lint/blob/master/lib/get-config.js#L15). (I'm assuming this is wherever atom was started from or maybe where the node executable is?).

Basically we're fine but there's an opportunity to tidy up our API usage.
